staging/src/k8s.io/apiserver/pkg/storage/etcd/etcd_helper.go

```diff
@@ -407,7 +407,7 @@ func (h *etcdHelper) decodeNodeList(nodes []*etcd.Node, filter storage.FilterFun
 			}
 			// being unable to set the version does not prevent the object from being extracted
 			_ = h.versioner.UpdateObject(obj, node.ModifiedIndex)
-			if filter(obj) {
+			if matched, _ := filter.Matches(obj); matched {
```

Comment:
That is potentially changing the semantic in case of error (if function returns true, err, then it will be true, previously being false).

In general, we shouldn't ignore errors (both here an below).

Reply:
Good catch. @wojtek-t Thanks for pointing out.